playbooks/provisioning/openstack/pre-install.yml

```diff
@@ -0,0 +1,72 @@
+---
+- hosts: cluster_hosts
+  become: true
+  pre_tasks:
+  - include: pre_tasks.yml
+  roles:
+  - role: hostnames
+
+# Subscribe Hosts
+- hosts: cluster_hosts
+  become: true
+  roles:
+  - role: subscription-manager
+    when: hostvars.localhost.rhsm_register
+    tags: 'subscription-manager'
+    ansible_sudo: true
+    retries: 5
+
+# Build DNS Records
+- hosts: localhost
+  pre_tasks:
+  - include: pre_tasks.yml
+  - name: "Generate dns-server views"
+    include: dns_dual_view.yml
+  - name: "Generate dns records"
+    include: dns_records.yml
+
+# Workaround - copy the DNS files over to the DNS server for sourcing ...
+- hosts: dns
+  become: true
+  tasks:
+  - name: "Copy named_views.yml"
+    copy:
+      src: /tmp/named_views.yml
+      dest: /tmp/named_views.yml
+  - name: "Copy records.yml"
+    copy:
+      src: /tmp/records.yml
+      dest: /tmp/records.yml
+
+# Configure DNS Server(s)
+- hosts: dns
+  become: true
+  pre_tasks:
+  - name: "Include the generated views"
+    include_vars: /tmp/named_views.yml
+  - name: "Include generated dns records"
+    include_vars: /tmp/records.yml
+  roles:
+    - role: dns-server
+    - role: dns
+
+# OpenShift Pre-Requisites
+- hosts: OSEv3
+  become: true
+  tasks:
+  - name: "Edit /etc/resolv.conf on masters/nodes"
```

Comment:
this should be persisted by network manager I believe, see https://github.com/openshift/openshift-ansible-contrib/issues/455